### DISCUSSION TITLE
Slow down test?

### DISCUSSION THREAD

**Discussion by pandoras-toolbox** [leancodepl/patrol#1549] [Q&A]
Is it possible to slow down a Patrol test without adding lots of sleep statements? It is too fast sometimes to follow it when observing.

I mean is there something in Patrol itself how it can be easily done?

Otherwise I might do it with some frameworking, proxying Patrol functions or something like that and adding the ability to slow down test execution in a generic central way.

**Comment by fylyppo** [comment]
There isn't any option in config to do so. The only way to globally wait between each actions is to have `PatrolTesterConfig(settlePolicy: SettlePolicy.settle)` or  `PatrolTesterConfig(settlePolicy: SettlePolicy.trySettle)` (it waits for all animations to finish to execute next action) but it's default so I assume you're using it.

But it's okay to play with overriding methods like:
```dart
void patrol(
  String description,
  Future<void> Function(OwnPatrolTester) callback,
) {
  patrolTest(
    description,
    nativeAutomation: true,
    ($) async {
      final ownPatrolTester$ = OwnPatrolTester(
        tester: $.tester,
        nativeAutomator: $.nativeAutomator,
        config: $.config,
      );

      await callback(ownPatrolTester$);
    },
  );
}

class OwnPatrolTester extends PatrolTester {
  OwnPatrolTester({
    required super.tester,
    required super.nativeAutomator,
    required super.config,
  });

  @override
  Future<void> tap(
    Finder finder, {
    bool? andSettle,
    SettlePolicy? settlePolicy,
    Duration? visibleTimeout,
    Duration? settleTimeout,
  }) async {
    await pumpAndTrySettle(duration: Duration(seconds: 1));
    return super.tap(
      finder,
      andSettle: andSettle,
      settlePolicy: settlePolicy,
      visibleTimeout: visibleTimeout,
      settleTimeout: settleTimeout,
    );
  }
}
```

**Comment by bartekpacia** [comment]
@pandoras-toolbox You could create your own `PandorasPatrolTester`, which would forward methods to `PatrolTester`, but add e.g. 0.5s delay before and after all actions.

FYI There's also a thing called [timeDilation](https://api.flutter.dev/flutter/scheduler/timeDilation.html). You may want to check that out. Disclaimer: I've never tried it with Patrol.

**Comment by pandoras-toolbox** [comment]
Thank you! I will try that out and report how it worked.
